From 29936d1f0c784e5350093bc60d9e4f45fa9fe5a3 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 12 Apr 2022 14:36:43 +0200
Subject: [PATCH] do not strictly depend on msginit

This restores pre-meson 0.62 behavior and fixes nls when
gettext-tiny is used.
---
 mesonbuild/modules/i18n.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/mesonbuild/modules/i18n.py b/mesonbuild/modules/i18n.py
index 09703bb..db4a978 100644
--- a/mesonbuild/modules/i18n.py
+++ b/mesonbuild/modules/i18n.py
@@ -226,7 +226,7 @@ class I18nModule(ExtensionModule):
             if self.tools[tool] is None:
                 self.tools[tool] = state.find_program(tool, required=False, for_machine=mesonlib.MachineChoice.BUILD)
             # still not found?
-            if not self.tools[tool].found():
+            if not self.tools[tool].found() and tool != 'msginit':
                 self.nogettext_warning(state.current_node)
                 return ModuleReturnValue(None, [])
         packagename = args[0]
@@ -295,7 +295,10 @@ class I18nModule(ExtensionModule):
         if extra_arg:
             updatepoargs.append(extra_arg)
         for tool in ['msginit', 'msgmerge']:
-            updatepoargs.append(f'--{tool}=' + self.tools[tool].get_path())
+            tool_path = self.tools[tool].get_path()
+            if not tool_path:
+                continue
+            updatepoargs.append(f'--{tool}=' + tool_path)
         updatepotarget = build.RunTarget(packagename + '-update-po', updatepoargs, [], state.subdir, state.subproject)
         targets.append(updatepotarget)
 
-- 
2.35.1

